### DISCUSSION TITLE
Is it working with Availability Groups?

### DISCUSSION THREAD

**Discussion by soedro** [trimble-oss/sql-log-shipping-service#96] [Q&A] [answered]
Hi,

Our current setup has a source db in an AG.
How would it work with sql-log-shipping-service?

Would somehow sending log files from all AG member instances to the same folder be the only way to do it, or you are doing it some other way?

Thanks.

**Comment by DavidWiseman** [comment]
If the location of your backups changes depending on what server is primary in the AG, that might complicate your restore process. Using a common UNC path would simplify things.  We use a file gateway for our backups in AWS with a common UNC path specified for backups with Ola's scripts.  Or direct to a common azure blob path in Azure.
The log shipping tool doesn't currently have the option to work with multiple directories.  It might be possible to support this in the future though.
If you have more than one AG and each AG has a different backup path, multiple instances of the log shipping tool could be used in this case.  That might also be a workaround for the different directories issue but I'm not sure how well it will work in that scenario.   If you wanted multiple instances of the log shipping tool deployed on the same computer, you can specify a different service name in the `sc create` step.

Hope this helps

**Comment by DavidWiseman** [comment]
In version 1.6 you can now specify a comma separated list of paths.  This is currently only supported for UNC paths - it doesn't work with Azure blob or S3.

**Reply by soedro** [reply]
Thanks for this change.
However, we need it to work with Azure blob location.

I am thinking of sending log files from all instances to the single blob container. That would be probably doable, but not as nice as if you would make it working like above, comma separated list of paths :-)

Thanks again!

**Reply by DavidWiseman** [reply, marked as answer]
Hi, it works with azure blob containers now.  It would need to be a single blob container but it supports a comma-separated list of paths.  Supporting multiple containers is harder as each would need a separate SAS token, but hopefully this works for you.

**Reply by soedro** [reply]
Thank you David, that is what we need.
I will test it and let you know how it goes.

Have a good one!

**Reply by soedro** [reply]
We are sending LOG files now to a single blob container and it is working that way.
Tested with failover and ungraceful shut down too. All good.

Thank you David on this tool, it is great!

**Reply by DavidWiseman** [reply]
Great!  Thanks for the feedback!
